fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java

```diff
@@ -631,22 +627,33 @@ public long getPrimaryComputeNodeIdByShard(long shardId, long workerGroupId) thr
         return backendIds.iterator().next();
     }
 
-    public List<Long> getAllNodeIdsByShard(long shardId, long workerGroupId, boolean onlyPrimary)
+    public long getPrimaryComputeNodeIdByShard(ShardInfo shardInfo) throws StarRocksException {
+        List<Long> ids = getAllNodeIdsByShard(shardInfo);
+        if (ids.isEmpty()) {
+            // If BE stops, routine load task may catch UserException during load plan,
+            // and the job state will changed to PAUSED.
+            // The job will automatically recover from PAUSED to RUNNING if the error code is REPLICA_FEW_ERR
+            // when all BEs become alive.
+            throw new StarRocksException(InternalErrorCode.REPLICA_FEW_ERR,
+                    "Failed to get primary backend. shard id: " + shardInfo.getShardId());
+        }
+        return ids.iterator().next();
```

Comment:
add comment to explain why pickup the first node as primary

Reply:
not in this pr, this is just a refactor